c++/src/H5Location.cpp

```diff
@@ -1834,16 +1833,20 @@ H5Location::getLinkval(const char *name, size_t size) const
     // if link has value, retrieve the value, otherwise, return null string
     if (val_size > 0) {
         // Create buffer for C string
-        value_C = new char[val_size + 1]();
+        value.assign(val_size, '\0');
 
-        ret_value = H5Lget_val(getId(), name, value_C, val_size, H5P_DEFAULT);
+        ret_value = H5Lget_val(getId(), name, &value[0], val_size, H5P_DEFAULT);
```

Comment:
This will only work for HDF5 1.13 and later because C++11 is required since 1.13. Without the use of C++11, one cannot guarantee that the C++ string is stored contiguously, so passing &value[0] will cause undefined behavior.